src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -74,6 +74,7 @@ public partial class SocketAsyncEventArgs : EventArgs, IDisposable
         private Socket? _currentSocket;
         private bool _userSocket; // if false when performing Connect, _currentSocket should be disposed
         private bool _disposeCalled;
+        internal Activity? _activity;
```

Comment:
```suggestion
        internal Activity? _connectActivity;
```

Reply:
Since there was some discussion around whether we need the field or not, if we expect that having the activity would be rare, you could e.g. stash it in a `ConditionalWeakTable`.

(I don't know that it matters more here than e.g. in HTTP though)

Reply:
> having the activity would be rare

In .NET 9 where the feature is experimental, extremely rare.

> I don't know that it matters more here than e.g. in HTTP though

This penalizes all Socket code that uses `SocketAsyncEventArgs` directly, including send/receive path. Contra-argument: perf-critical code, such as Kestrel transport should reuse such `SocketAsyncEventArgs` anyways.

@stephentoub should I look into how to get rid of the field using `ConditionalWeakTable`?

Reply:
The field on SAEA is there purely to know which Activity to use in post-Connect logging? Using a CWT will mean a dictionary lookup after every connect, even if there is no activity for the SAEA instance. We could lazily initialize the CWT itself if the common case will be no requests in the process ever have an activity.

Reply:
> The field on SAEA is there purely to know which Activity to use in post-Connect logging?

Yes.

> We could lazily initialize the CWT itself if the common case will be no requests in the process ever have an activity.

This sounds good enough for 9. Later, maybe we can merge a some status fields, introducing a bit indicating if there is an Activity?